File: 2022-1-elixir/test/day_07_test.exs
defmodule Day07Test do
  use ExUnit.Case

  setup_all do
    IO.puts("--- DAY 07 ----")
    {:ok, test_data} = File.read("data/day_07_test.txt")
    {:ok, real_data} = File.read("data/day_07.txt")
    {:ok, test_data: test_data, real_data: real_data}
  end

  describe "part 1" do
    test "test data", context do
      test_result = Day07.part_1(context[:test_data])
      IO.puts("part 1 :: test : #{test_result}")
      assert test_result == 95_437
    end

    test "real data", context do
      real_result = Day07.part_1(context[:real_data])
      IO.puts("part 1 :: real : #{real_result}")
      assert real_result
      assert real_result == 1_084_134
    end
  end

  describe "part 2" do
    test "test data", context do
      test_result = Day07.part_2(context[:test_data])
      IO.puts("part 2 :: test : #{test_result}")
      assert test_result == 24_933_642
    end

    test "real data", context do
      real_result = Day07.part_2(context[:real_data])
      IO.puts("part 2 :: real : #{real_result}")
      assert real_result
      assert real_result == 6_183_184
    end
  end
end

File: 2024/day_07_test.rb
# frozen_string_literal: true

require './test_helper'
require './day_07'

describe Day07 do
  before do
    @data = StringIO.new(<<~TEXT)
    190: 10 19
    3267: 81 40 27
    83: 17 5
    156: 15 6
    7290: 6 8 6 15
    161011: 16 10 13
    192: 17 8 14
    21037: 9 7 18 13
    292: 11 6 16 20
    TEXT
  end

  it 'returns 3_749 for part 1' do
    assert_equal 3_749, Day07.new(io: @data).part_1!
  end

  it 'returns 11_387 for part 2' do
    assert_equal 11_387, Day07.new(io: @data).part_2!
  end
end
